MiniApp/Classes/core/RealMiniApp.swift

```diff
@@ -205,13 +240,69 @@ internal class RealMiniApp {
                                               miniAppVersion: version,
-                                              apiClient: self.miniAppClient, completionHandler: self.createCompletionHandler(completionHandler: completionHandler))
+                                              apiClient: self.miniAppClient) { (result) in
+            switch result {
+            case .success(let metaData):
+                self.miniAppManifest = metaData
+                completionHandler(.success(metaData))
+            case .failure(let error):
+                completionHandler(.failure(.fromError(error: error)))
+            }
+        }
     }
 
     /// Method to remove the unused/deleted items from the Keychain
     func cleanUpKeychain() {
         self.miniAppStatus.removeUnusedCustomPermissions()
     }
+
+    /// Method to check if all the required permissions mentioned in the manifest.json is agreed by the user.
+    /// - Parameters:
+    ///   - appId: MiniApp ID
+    ///   - versionId: Specific VersionID of a MiniApp
+    ///   - completionHandler: Handler that returns whether user agreed to required permissions or not.
+    func isRequiredPermissionsAllowed(appId: String, versionId: String, completionHandler: @escaping (Result<Bool, MASDKError>) -> Void) {
+        retrieveMiniAppMetaData(appId: appId, version: versionId) { (result) in
+            switch result {
+            case .success(let manifest):
+                guard let requiredPermissions = manifest.requiredPermissions, requiredPermissions.count > 0 else {
+                    completionHandler(.success(true))
+                    return
+                }
+                self.verifyRequiredPermissions(appId: appId,
+                                               requiredPermissions: requiredPermissions,
+                                               completionHandler: completionHandler)
+            case .failure(let error):
+                completionHandler(.failure(.fromError(error: error)))
+            }
+        }
+    }
+
+    /// Method that compares the required permissions from the manifest and the stored custom permissions.
+    /// - Parameters:
+    ///   - appId: MiniApp ID
+    ///   - requiredPermissions: List of required Custom permissions that is defined by the Mini App
+    ///   - completionHandler: Handler that returns whether user agreed to required permissions or not.
+    func verifyRequiredPermissions(appId: String,
+                                   requiredPermissions: [MASDKCustomPermissionModel],
+                                   completionHandler: @escaping (Result<Bool, MASDKError>) -> Void) {
+        let storedCustomPermissions = self.miniAppPermissionStorage.getCustomPermissions(forMiniApp: appId)
+        let filtered = storedCustomPermissions.filter {
+            requiredPermissions.contains($0)
+        }
+        if filtered.count == requiredPermissions.count && filtered.allSatisfy({ $0.isPermissionGranted.boolValue == true }) {
```

Comment:
I thing you could add the `allSatisfy` _(awesome language feature!)_ condition to the filter above

Reply:
No, we shouldn't. Stored Permissions contains both required and optional permissions So we need to filter only required permissions and make sure only they have all permissions `.allowed`